pkg/repo/git.go

```diff
@@ -0,0 +1,55 @@
+package repo
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
+)
+
+// HTTPClient defines an interface for HTTP get requests.
+//go:generate counterfeiter -o fakes/fake_http_client.go . HTTPClient
+type HTTPClient interface {
+	Get(string) (*http.Response, error)
+}
+
+var httpClient HTTPClient = http.DefaultClient
+
+// GetProfileDefinition returns a definition based on a url and a branch.
+func GetProfileDefinition(repoURL, branch, path string) (profilesv1.ProfileDefinition, error) {
```

Comment:
I know this is copied code, but I really really would prefer if that httpClient wouldn't be a package local variable. :D But rather would hang in a struct and this profile would have a struct receiver. Would be easier to mock as well, I believe.

Reply:
yeah I don't feel strongly either way. I like the pattern of `export_test.go` to "hide" the mocking from the consumer (who wouldn't care what httpClient was passed in). Having it on a struct instead and set the same way is fine, although a struct with 1 func might as well just be a func :smile:

Reply:
:D, true. :)